### DISCUSSION TITLE
How to turn on LSP with Mason?

### DISCUSSION THREAD

**Discussion by levaspb** [NvChad/NvChad#1435] [Q&A] [answered]
I am newbie in Neovim, so I spent about hour finding the way to turn on YAML language server which I installed with Mason. Still it doesn’t work as expected. I suspect that problem is this: Lspconfig was replaced with Mason and no any written instructions what to do. I would like to turn on usual features like autocompletion, syntax checking and so on for Kubernetes YAML files. So the second question is how to choose right schema for LSP. I would be very greatful for detailed answear.

**Comment by siduck** [comment, marked as answer]
@levaspb mason just downloads the lsp servers, formatters etc. you need to setup custom lspconfig 

https://nvchad.com/docs/config/lsp/

**Reply by levaspb** [reply]
Thank you but I read it before. There is no clear instructions how should I setup lspconfig there. So I followed another way. I added fragment described [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#yamlls) to lspserver.lua file. I am not sure this is correct but now nvim starts language server when I am opening yaml files.

**Reply by siduck** [reply]
@levaspb i see you are new to nvchad so please read the docs from the start to end sequentially, they are very important. You cant read specific parts of docs unless you're an experienced nvchad user

**Reply by gkobeaga** [reply]
The link for configuring LSP in NvChad has changed. You can now find the latest docs in **[NvChad LSP Custom Configuration](https://nvchad.com/docs/config/lsp)**.

**Comment by m4ss1m0g** [comment]
I had the same difficulties; the linked docs are useful but cryptic for a newbie, indeed, you MUST read the README on [spconfigure repository](https://github.com/neovim/nvim-lspconfig).

Basically you must:

1. Launch the `Mason` and install the LSP
2. GoTo the [Config.md](github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md) indicated from the NvChad doc and search the LSP
3. Open the `~/.config/nvim/lua/configs/lspconfig.lua` and add the code below
4. Optionally tweaks the configuration (refer to the configs.md aforementioned

``` ini
require("nvchad.configs.lspconfig").defaults()

-- added omnisharp
local servers = { "html", "cssls", "omnisharp" }
vim.lsp.enable(servers)

-- read :h vim.lsp.config for changing options of lsp servers 
--

-- configure omnisharp
vim.lsp.config('omnisharp', {
  settings = {
    FormattingOptions = {
      OrganizeImports = true
    },
    RoslynExtensionsOptions = {
      EnableAnalyzersSupport = true
    }
  },

})

```
